package/MDAnalysis/core/universe.py

```diff
@@ -372,6 +372,13 @@ def __init__(self, topology=None, *coordinates, all_coordinates=False,
 
         if guess_bonds:
             self.atoms.guess_bonds(vdwradii=vdwradii)
+        # add mass and type to the to_guess list
+        toplist = list(self._topology.read_attributes)
+        if not any(att.singular == 'type' for att in toplist) and 'type' not in to_guess:
```

Comment:
The next few lines look as if they could be written as a function so that we don’t have to copy and paste when we guess more than just mass and type

Reply:
I just added them for handling the current behavior of automatically guessing mass and types, which I moved out of parsers. Checking other guessing attributes is dealt with in the `guess_TopologyAttributes` method